packages/server/graphql/mutations/resetRetroMeetingToGroupStage.ts

```diff
@@ -56,43 +75,31 @@ const resetMeetingToStage = {
       return standardError(new Error('The meeting has already ended'), {userId: viewerId})
 
     // RESOLUTION
-    // TODO don't create all the phases from scratch
-    const createdPhases = await createNewMeetingPhases(
-      viewerId,
-      teamId,
-      meetingId,
-      meetingCount,
-      meetingType,
-      dataLoader
-    )
-    const discussionIds = [] as string[]
-    let shouldResetStage = false
+    let discussionIds = [] as string[]
     let resetToPhaseIndex = -1
     const newPhases = [] as GenericMeetingPhase[]
     for (const [phaseIndex, phase] of phases.entries()) {
```

Comment:
this feels a little too complex.
let's avoid mutating old objects & having a phaseType conditional (resetMeetingPhase) inside a phaseType switch/case.

```ts
    const newPhases = phases.map((phase) => {
      switch (phase.phaseType) {
        case 'checkin':
        case 'reflect':
          return phase
        case 'group':
        case 'vote':
          return new GenericMeetingPhase(phase.phaseType)
        case 'discuss':
          return new DiscussPhase([])
      }
    })

    const discussionIdsToDelete = phases.find((phase) => phase.phaseType === 'discuss')?.stages.map(({discussionId}) => discussionId)
    
```